src/builder.rs

```diff
@@ -130,7 +144,17 @@ impl ActionInfo {
     }
 
     /// Returns the value sum for this action.
+    /// Split notes does not contribute to the value sum.
     fn value_sum(&self) -> ValueSum {
+        // TODO: Aurel, uncomment when circuit for split flag is implemented.
+        // let spent_value = self
+        //     .spend
+        //     .split_flag
+        //     .then(|| self.spend.note.value())
+        //     .unwrap_or_else(NoteValue::zero);
```

Comment:
This appears to be the wrong way round: the value should be included iff `split_flag` is false.

Reply:
Actually, the code in question does precisely that. I will refactor it as suggested for more clarity.